pkg/scene/widget_align_system.go

```diff
@@ -0,0 +1,200 @@
+package scene
+
+import (
+	"github.com/reearth/reearth-backend/pkg/id"
+)
+
+// WidgetAlignSystem is the layout structure of any enabled widgets that will be displayed over the scene.
+type WidgetAlignSystem struct {
+	inner WidgetZone
+	outer WidgetZone
+}
+
+// WidgetZone is the structure of each layer (inner and outer) of the align system.
+type WidgetZone struct {
+	left   WidgetSection
+	center WidgetSection
+	right  WidgetSection
+}
+
+// WidgetSection is the structure of each section of the align system.
+type WidgetSection struct {
+	top    WidgetArea
+	middle WidgetArea
+	bottom WidgetArea
+}
+
+// WidgetArea has the widgets and alignment information found in each part area of a section.
+type WidgetArea struct {
+	widgetIds []*id.WidgetID
+	align     string
+}
+
+type WidgetLocation struct {
+	Zone    string
+	Section string
+	Area    string
+}
+
+// NewWidgetAlignSystem returns a new widget align system.
+func NewWidgetAlignSystem() *WidgetAlignSystem {
+	return &WidgetAlignSystem{}
+}
+
+// WidgetZone will return a specific zone in the align system.
+func (was *WidgetAlignSystem) WidgetZone(zone string) *WidgetZone {
```

Comment:
```suggestion
func (was *WidgetAlignSystem) Zone(zone string) *WidgetZone {
```

You can rather shorten the name here, since the only possible Zone of WidgetAlignSystem is widget's one, so let's keep the method name clear.